tests/conftest.py

```diff
@@ -1,4 +1,23 @@
 import sys
 from pathlib import Path
+from unittest.mock import patch
+
+import pytest
 
 sys.path.insert(0, str(Path(__file__).parent.parent))
+
+
+@pytest.fixture(scope="function")
+def generated_rows(request):
+    p = patch("snowfakery.output_streams.DebugOutputStream.write_single_row")
+    mockobj = p.start()
+
+    def row_values(index, value):
+        return mockobj.mock_calls[index][1][1][value]
+
+    mockobj.row_values = row_values
+    yield mockobj
+
+    p.stop()
+
+    return
```

Comment:
The return is a no-op, isn't it?

You also could use patch as a context manager and avoid the explicit calls to start/stop. That would help avoid the bug here where stop doesn't get called if the test raises an exception

Reply:
744b9bb